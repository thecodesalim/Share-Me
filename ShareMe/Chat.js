import React from "react";
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  KeyboardAvoidingView,
  SafeAreaView,
  Image
} from "react-native";

import { withGlobalContext } from "./SocketContext";

export class Chat extends React.Component {
  constructor(props) {
    super(props);
  }
  state = {
    messages: [],
    chatMessage: ""
  };

  componentDidMount() {
    this.props.socket.on("chat", msg =>
      this.setState({ messages: [...this.state.messages, msg] })
    );
  }

  onSubmitMessage(e) {
    console.log(e + " fukkkk");
    this.props.socket.emit("chat", this.state.chatMessage);
    this.setState({ chatMessage: "" });
    console.log(this.state.message);
  }
  render() {
    const m = this.state.messages.map(i => (
      <Text key={i} style={styles.bubble}>
        {i}
      </Text>
    ));
    return (
      <KeyboardAvoidingView style={{ flex: 1 }} behavior="padding">
        <SafeAreaView style={styles.container}>
          <View style={styles.inner}>
            <Text>Dave</Text>
            {m}
          </View>
          <View style={styles.send}>
             <Text style={{padding: 10}}>Hello</Text>
            </View>
            <View style={styles.receive}>
            <Text style={{padding: 10}}>Hi</Text>
            </View>
          <View style={styles.position}>
            <TextInput
              style={{
                height: 40,
                width: 300,
                borderColor: "gray",
                borderWidth: 0.5,
                borderRadius: 20,
              }}
              placeholder="message"
              value={this.state.chatMessage}
              onSubmitEditing={e => this.onSubmitMessage(e)}
              onChangeText={chatMessage => this.setState({ chatMessage })}
            />
            <Image
              source={require('./assets/send.png')}
            />
          </View>
        </SafeAreaView>
      </KeyboardAvoidingView>
    );
  }
}

export default withGlobalContext(Chat);

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  inner: {
    //padding: 24,
    flex: 1,
    justifyContent: "flex-start",
    alignSelf: "center"
  },
  send: {
    height: 40,
    width: 100,
    margin: 10,
    backgroundColor: "rgba(236, 100, 75, 0.5)",
    alignSelf: "flex-start",
    //marginTop: 10
  },
  receive: {
    height: 40,
    width: 100,
    margin: 10,
    backgroundColor: "rgba(236, 100, 75, 0.5)",
    alignSelf: "flex-end",
    
   // marginTop: 10
  },
  position: {
    flex: 1,
    justifyContent: "flex-end",
    flexDirection: "row",
    alignSelf: "center",
    //marginTop: 200
  },
  bubble: {
    height: 40,
    width: 100,
    backgroundColor: "tomato",
    alignSelf: "center",
    marginTop: 10,
    borderRadius: 20,
  },
  
});
