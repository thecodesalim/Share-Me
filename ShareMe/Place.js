import React from "react";
import { StyleSheet, Text, View, ScrollView } from "react-native";
import { MapView, Location, Permissions, Marker, Callout } from "expo";

export default class Place extends React.Component {


  render() {
    return (
      <View>
        <MapView.Marker
          coordinate={{
            latitude: this.props.latitude,
            longitude: this.props.longitude
          }}
        >
          <View style={this.props.circle} />
          <MapView.Callout tooltip={true} style={styles.box}
          onPress={() => this.props.onTap}
          >
            <View>
              <Text
                style={{ fontSize: 13, fontWeight: "600", marginBottom: 10 }}>
                {this.props.title}
              </Text>
              <ScrollView>{this.props.description}</ScrollView>
            </View>
          </MapView.Callout>
        </MapView.Marker>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  box: {
    width: 300,
    height: 150,
    borderRadius: 10,
    backgroundColor: "white",
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
    shadowColor: "blue",
    padding: 10
  }
});
