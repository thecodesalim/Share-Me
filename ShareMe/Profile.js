import React from 'react'
import { StyleSheet, Text, View, Button} from 'react-native'

export default class Profile extends React.Component {
    render(){
        return(
            <View style={styles.container}>
                <View style={styles.box}>
                    <Text style={{margin: 10, paddingTop: 20}}>Salim Abubakar</Text>
                    <Text>Portsmouth</Text>
                    <View style={styles.button}><Text style={styles.text}>Edit Profile</Text></View>
                </View>
            </View>
        )
    }
}

const styles = StyleSheet.create({

    container: {
        flex: 1
    },

    box:{
        marginTop: 20,
        width: '100%',
        height: '40%',
        backgroundColor: '#EDEDED',
        flexDirection: 'column',
        alignItems: 'center',
        shadowColor: 'blue',
        padding: 10,
    },

    text: {
        marginTop: 10,
        borderRadius: 12,
        borderColor: 'black',
        color: 'black',
        fontWeight: 'bold',
        overflow: 'hidden',
        padding: 12,
        textAlign:'center',
    },
    button: {
        marginTop: 200,
        margin: 20,
        borderStyle: 'solid',
        borderWidth: 1,
        width: 100 ,
        backgroundColor: 'white',
        height: 50,
        borderRadius: 12,
        borderColor: 'black',
    }
  })