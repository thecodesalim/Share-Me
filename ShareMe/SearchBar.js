import React from 'react'
import { StyleSheet, Text, View, TextInput, ScrollView} from 'react-native'
import { MapView, Location, Permissions, Marker, Callout } from 'expo'


export default class SearchBar extends React.Component {
    render(){
        return (
            <View style={styles.bar}>
                
                <View style={{paddingTop:20}}> 
                <TextInput style={{height: 40, borderColor: 'gray', borderWidth: 1}}/>
                </View>
                
            </View>
        )
    }
}

const styles = StyleSheet.create({
   bar: {
       width: 300,
       height: 100,
       flexDirection: 'column',
       backgroundColor: 'pink'
   }
  })

  //  AIzaSyBwRCodJDLOr0js_sl88pW96tsqGvyxpKM 