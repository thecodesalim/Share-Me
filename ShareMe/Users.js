import React from 'react'
import { StyleSheet, Text, View} from 'react-native'

export default class Users extends React.Component {
  render(){
    return(
        <View style={styles.member} >
            <View style={styles.circle}></View>
            <Text>{this.props.name}</Text>     
        </View>
      
    )
  }
}

const styles = StyleSheet.create({
    calloutView: {
        flexDirection: "row",
        backgroundColor: "rgba(255, 255, 255, 0.9)",
        borderRadius: 10,
        width: "80%",
        marginLeft: "30%",
        marginRight: "30%",
        marginTop: 20
      },
      circle: {
        width: 30,
        height: 30,
        borderRadius: 30 / 2,
        backgroundColor: 'yellow',
    },
    member: {
        flexDirection: 'row',
        justifyContent: 'space-around',
        alignItems: 'flex-start',
        backgroundColor: 'rgba(255,255,255,0.5)',
        borderRadius: 20,
        height: 30,
        marginTop: 10,
        borderColor: "transparent",
        marginLeft: 10,
        width: "90%",
        marginRight: 10,
        height: 40,
        borderWidth: 0.0 , 
        paddingTop: 5
    },
});
