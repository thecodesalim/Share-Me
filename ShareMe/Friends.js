import React from "react";
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableOpacity
} from "react-native";
import { createStackNavigator, createAppContainer } from "react-navigation";
import Chat from "./Chat";

export class FriendsView extends React.Component {
  render() {
    return (
      <View>
        <TouchableOpacity 
        style={styles.box}
        onPress={() => this.props.navigation.navigate('Chat')}>
          <Text> Dave </Text>
        </TouchableOpacity>
        <TouchableOpacity 
        style={styles.box}
        onPress={() => this.props.navigation.navigate('Chat')}>
          <Text> Sury </Text>
        </TouchableOpacity>
        
      </View>
    );
  }
}

const RootStack = createStackNavigator(
  {
    Home: FriendsView,
    Chat: Chat
  },
  {
    initialRouteName: "Home"
  }
);

export default AppContainer = createAppContainer(RootStack);

//export const AppContainer = createAppContainer(RootStack);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20
  },

  box: {
    marginBottom: 5,
    width: "100%",
    height: 75,
    backgroundColor: "#F5F5F5",
    alignSelf: "center",
    justifyContent: "center",
    textAlign:'center',
  }
});
