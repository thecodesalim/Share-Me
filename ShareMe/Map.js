import React from "react";
import { StyleSheet, Text, View, TextInput, AlertIOS } from "react-native";
import { MapView, Location, Permissions } from "expo";
import Place from "./Place";
import Users from "./Users";
//window.navigator.userAgent = "react-native";
import { withGlobalContext } from "./SocketContext";

export class Map extends React.Component {
  constructor(props) {
    super(props);
  }
  static navigationOptions = {
    header: null
  };

  state = {
    location: null,
    policeData: [],
    desc: "",
    show: false,
    usersLocation: [],
    message: [],
    users: []
  };

  componentDidMount() {
    this.getUserLocation();
    this.props.socket.on("chat", location =>
      this.setState({ usersLocation: [...this.state.usersLocation, location] })
    );

    AlertIOS.prompt("Please type your name", null, name =>
      this.props.socket.emit("name", name)
    );
    this.getUsers();
  }

  getUserLocation = async () => {
    let { status } = await Permissions.askAsync(Permissions.LOCATION);
    if (status !== "granted") {
      this.setState({
        errorMessage: "Permission to access location was denied"
      });
    }
    let location = await Location.getCurrentPositionAsync({});
    this.setState({
      location: location
    });
    const position = location.coords;
    this.getData(position.latitude, position.longitude);
  };

  onlyUnique = (value, index, self) => {
    return self.indexOf(value) === index;
  };

  getData = async (lat, long) => {
    //const response = await fetch(`https://data.police.uk/api/crimes-street/all-crime?lat=${lat}&lng=${long}`)

    const response = await fetch(
      `https://data.police.uk/api/crimes-at-location?lat=${lat}&lng=${long}`
    );
    
    const data = await response.json();

    let policeData = data.map(i => i);
    this.setState({
      policeData: this.state.policeData.concat(policeData)
    });
  };

  onSubmitMessage = () => {
    this.props.socket.emit("chat", this.state.location);
    console.log("come on please work, YA ALLAH");
  };

  getUsers = () => {
    this.props.socket.on("name", id => {
      this.setState({
        users: [...this.state.users, id]
      });
    });
  };

  getCrimeDetails = () => {
    console.log('i love and hate programming')
  }

  createHeatMap = () => {
    let color = "rgba(255, 236, 139, 0.1)"
    this.state.policeData.map( i => {
      if(i.category === "violent-crime") {
        color = 'red'
      }
      else { 
        color = 'pink'
      }
      return(
        <MapView.Marker 
        center={{
          latitude: Number(i.location.latitude),
          longitude: Number(i.location.longitude),
        }}
        radius={25}
        fillColor={color}
        strokeColor= "rgba(255, 236, 139, 0.1)"
        >

        </MapView.Marker>
      )
    })
  }

  render() {
    if (!this.state.location) {
      return (
        <View>
          <Text>Loading Location</Text>
        </View>
      );
    }

    if (!this.state.policeData) {
      return (
        <View>
          <Text>Loading Location</Text>
        </View>
      );
    }

  
      let color = "rgba(255, 236, 139, 0.1)"
      this.state.policeData.map( i => {
        if(i.category === "violent-crime") {
          color = "rgba(236, 100, 75, 0.1)"
        }
        else { 
          color = "rgba(255, 236, 139, 0.1)"
        }
      })

    let markers = [];
    this.state.policeData.map(i => markers.push(i));
    let users = this.state.users.map(i => <Users name={i} key={i} />);
    this.state.users.map(i => console.log(i));
  
    return (
      <View style={styles.container}>
        <MapView
          onPress={(e) => console.log(e)}
          provider={MapView.PROVIDER_GOOGLE}
          style={styles.map}
          region={{
            latitude: this.state.location.coords.latitude,
            longitude: this.state.location.coords.longitude,
            latitudeDelta: 0.0922 / 50,
            longitudeDelta: 0.0421 / 50
          }}
        >
          <MapView.Circle
            center={this.state.location.coords}
            radius={50}
            fillColor="rgba(236, 100, 75, 0.2)"
            strokeColor ="rgba(236, 100, 75, 0.2)"
          />
          <MapView.Marker
            draggable
            coordinate={this.state.location.coords}
            title="You are here"
            description={this.state.desc}
            onDragEnd={e =>
              this.getData(
                e.nativeEvent.coordinate.latitude,
                e.nativeEvent.coordinate.longitude
              )
            }
          >
          </MapView.Marker>

          <MapView.Marker
            draggable
            coordinate={{
              latitude: 50.798400,
              longitude:-1.09147840
            }}
            title="Dave"
            description={this.state.desc}
            pinColor={'yellow'}
          ></MapView.Marker>

          {markers.map(i => {                                                              
            return (
              <View>
              <Place
                circle={styles.circle}
                key={i.persistent_id}
                latitude={Number(i.location.latitude)}
                longitude={Number(i.location.longitude)}
                title={i.category}
                description={<Text>{i.location.street.name}</Text>}
              />
              <MapView.Circle
                center={{
                  latitude: Number(i.location.latitude),
                  longitude: Number(i.location.longitude),
                }}
                radius={25}
                fillColor={color}
                strokeColor= "rgba(255, 236, 139, 0.1)"
              />
              </View>
              );
            
          })}
        </MapView>

        <MapView.Callout>
          <View style={styles.calloutView}>
            <TextInput
              style={styles.calloutSearch}
              placeholder={"Search"}
              onSubmitEditing={this.createHeatMap}
            />
          </View>
        </MapView.Callout>

        <MapView.Callout style={styles.members}>
          {users}
        </MapView.Callout>
      </View>
    );
  }
}

export default withGlobalContext(Map);

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  map: {
    flex: 1
  },
  box: {
    marginTop: 20,
    alignSelf: "center",
    width: 300,
    height: 80,
    borderRadius: 10,
    backgroundColor: "white",
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
    shadowColor: "blue",
    padding: 10
  },
  calloutView: {
    flexDirection: "row",
    backgroundColor: "rgba(255, 255, 255, 0.9)",
    borderRadius: 10,
    width: "40%",
    marginLeft: "30%",
    marginRight: "30%",
    marginTop: 20
  },
  calloutSearch: {
    borderColor: "transparent",
    marginLeft: 10,
    width: "90%",
    marginRight: 10,
    height: 40,
    borderWidth: 0.0
  },
  members: {
    flexDirection: "column",
    borderRadius: 10,
    width: "40%",
    marginLeft: "0%",
    marginRight: "0%",
    marginTop: 260
  },
  circle: {
    width: 30,
    height: 30,
    borderRadius: 30 / 2,
    backgroundColor: "red"
  },
  usercircle: {
    width: 30,
    height: 30,
    borderRadius: 30 / 2,
    backgroundColor: "lightblue"
  },
  pinText: {
    color: "white",
    fontWeight: "bold",
    textAlign: "center",
    fontSize: 20,
    marginBottom: 10
  }
});
