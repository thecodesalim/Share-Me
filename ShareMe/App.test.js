import React from "react";
import Place from "./Place";
import Map from "./Map";
import Profile from "./Profile";
import Chat from "./Chat";
import Users from "./Users";
import SearchBar from "./SearchBar";
import renderer from "react-test-renderer";
import { FriendsView } from "./Friends";

import { mount } from "enzyme";

test("Place Class renders correctly", () => {
  const tree = renderer.create(<Place />).toJSON();
  expect(tree).toMatchSnapshot();
});

test(" FriendsView Class renders correctly", () => {
  const tree = renderer.create(<FriendsView />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Map Class renders correctly", () => {
  const tree = renderer.create(<Map />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Chat Class renders correctly", () => {
  const tree = renderer.create(<Chat />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Place Class render user position correctly with null values", () => {
  const props = {
      latitude: null,
      longitude: null
    },
    location = mount(<Place {...props} />);
  expect(location.prop("latitude")).toEqual(null);
  expect(location.prop("longitude")).toEqual(null);
});

test("Place Class render user position correctly with valid values", () => {
  const props = {
      latitude: 1,
      longitude: 0.9
    },
    location = mount(<Place {...props} />);
  expect(location.prop("latitude")).toEqual(1);
  expect(location.prop("longitude")).toEqual(0.9);
});

test("User Class render user name correctly with null values", () => {
  const props = {
      name: ""
    },
    location = mount(<Users {...props} />);
  expect(location.prop("name")).toEqual("");
});

test("User Class render user name correctly with valid values", () => {
  const props = {
      name: "Human"
    },
    location = mount(<Users {...props} />);
  expect(location.prop("name")).toEqual("Human");
});

test("Profiile Class renders correctly", () => {
  const tree = renderer.create(<Profile />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Users Class enders correctly", () => {
  const tree = renderer.create(<Users />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("SearchBar Class renders correctly", () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Sending Message", () => {
  const location = mount(<Map />);
  const newTodoTextInput = location.find("TextInput").first();
  expect(location.find(Map)).toExist();
});

test("Update Location", () => {
  newTodoTextInput.props().onChangeText(newTodoText);

      addTodoButton.props().onPress();

      wrapper.update();
      expect(
        wrapper.findWhere(node => node.prop('testID') === 'todo-item'),
      ).toExist();
});

jest.useFakeTimers();
