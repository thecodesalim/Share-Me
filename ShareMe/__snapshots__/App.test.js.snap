// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` FriendsView Class renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#F5F5F5",
        "height": 75,
        "justifyContent": "center",
        "marginBottom": 5,
        "opacity": 1,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <Text>
        Salim 
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#F5F5F5",
        "height": 75,
        "justifyContent": "center",
        "marginBottom": 5,
        "opacity": 1,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <Text>
       Sury 
    </Text>
  </View>
</View>
`;

exports[`Chat Class renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "paddingTop": 50,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View>
    <Text>
      Name
    </Text>
  </View>
  <View
    style={
      Object {
        "justifyContent": "flex-end",
        "marginBottom": 20,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="message"
      style={
        Object {
          "borderColor": "gray",
          "height": 40,
          "width": 500,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;

exports[`Map Class renders correctly 1`] = `
<View>
  <Text>
    Loading Location
  </Text>
</View>
`;

exports[`Place Class renders correctly 1`] = `
<View>
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": undefined,
        "longitude": undefined,
      }
    }
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
  >
    <View />
    <AIRMapCallout
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "flex": 1,
            "flexDirection": "column",
            "height": 150,
            "padding": 10,
            "shadowColor": "blue",
            "width": 300,
          },
        ]
      }
      tooltip={true}
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 13,
              "fontWeight": "600",
              "marginBottom": 10,
            }
          }
        />
      </View>
    </AIRMapCallout>
  </AIRMapMarker>
</View>
`;

exports[`Profiile Class renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flexDirection": "column",
        "height": "40%",
        "marginTop": 20,
        "padding": 10,
        "shadowColor": "blue",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "blue",
          "borderRadius": 50,
          "height": 100,
          "marginBottom": 20,
          "width": 100,
        }
      }
    />
    <Text>
      Salim Abubakar
    </Text>
    <Text>
      Portsmouth
    </Text>
    <View>
      <Text
        style={
          Object {
            "backgroundColor": "blue",
            "borderRadius": 12,
            "color": "white",
            "fontWeight": "bold",
            "marginTop": 10,
            "overflow": "hidden",
            "padding": 12,
            "textAlign": "center",
          }
        }
      >
        Edit Profile
      </Text>
    </View>
  </View>
</View>
`;

exports[`SearchBar Class renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "pink",
      "flexDirection": "column",
      "height": 100,
      "width": 300,
    }
  }
>
  <View
    style={
      Object {
        "paddingTop": 20,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`Users Class enders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "rgba(255,255,255,0.5)",
      "borderColor": "transparent",
      "borderRadius": 20,
      "borderWidth": 0,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-around",
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingTop": 5,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "yellow",
        "borderRadius": 15,
        "height": 30,
        "width": 30,
      }
    }
  />
  <Text />
</View>
`;
